--- a/nss/lib/freebl/Makefile	2017-10-16 10:11:38.750787330 +0300
+++ b/nss/lib/freebl/Makefile	2017-10-16 10:11:56.629600056 +0300
@@ -108,15 +108,15 @@
 # NSS_X86 means the target is a 32-bits x86 CPU architecture
 # NSS_X64 means the target is a 64-bits 64 CPU architecture
 # NSS_X86_OR_X64 means the target is either x86 or x64
-ifeq (,$(filter-out i386 x386 x86 x86_64,$(CPU_ARCH)))
-        DEFINES += -DNSS_X86_OR_X64
-        CFLAGS += -mpclmul -maes
-ifneq (,$(USE_64)$(USE_X32))
-        DEFINES += -DNSS_X64
-else
-        DEFINES += -DNSS_X86
-endif
-endif
+#ifeq (,$(filter-out i386 x386 x86 x86_64,$(CPU_ARCH)))
+#        DEFINES += -DNSS_X86_OR_X64
+#        CFLAGS += -mpclmul -maes
+#ifneq (,$(USE_64)$(USE_X32))
+#        DEFINES += -DNSS_X64
+#else
+#        DEFINES += -DNSS_X86
+#endif
+#endif

 ifeq ($(OS_TARGET),OSF1)
     DEFINES += -DMP_ASSEMBLY_MULTIPLY -DMP_NO_MP_WORD
