--- /dev/null
+++ b/nss/coreconf/OpenWRT.mk
@@ -0,0 +1,117 @@
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+include $(CORE_DEPTH)/coreconf/UNIX.mk
+
+RANLIB = ranlib
+CC = $(TARGET)
+DEFAULT_COMPILER= $(TARGET)
+CCC = $(TARGETCXX)
+CROSS_COMPILE = 1
+NATIVE_CC = gcc
+DEFINES += -DHAVE_POSIX_FALLOCATE=0  -D_REENTRANT
+OPTIMIZER = -Os -gdwarf-2
+
+ifeq (,$(filter-out ppc64 ppc64le,$(CPU_ARCH)))
+	CPU_ARCH	= ppc
+ifeq ($(USE_64),1)
+	ARCHFLAG	= -m64
+endif
+else
+ifeq ($(CPU_ARCH),alpha)
+	OS_REL_CFLAGS   = -D_ALPHA_
+	CPU_ARCH	= alpha
+else
+ifeq ($(CPU_ARCH),x86_64)
+	CPU_ARCH	= x86_64
+	ARCHFLAG	= -m64
+	CFLAGS		+= -mpclmul -maes
+	DEFINES		+= -DNSS_X64 -DNSS_USE_64
+else
+ifeq ($(CPU_ARCH),i386)
+	OS_REL_CFLAGS	= -Di386
+	CPU_ARCH	= x86
+	ARCHFLAG	= -m32
+	CFLAGS		+= -mpclmul -maes
+	DEFINES		+= -DNSS_X86
+else
+ifeq ($(CPU_ARCH),sparc64)
+	CPU_ARCH        = sparc
+else
+ifeq (,$(filter-out arm% sa110,$(CPU_ARCH)))
+	CPU_ARCH        = arm
+else
+ifeq (,$(filter-out parisc%,$(CPU_ARCH)))
+	CPU_ARCH        = hppa
+else
+ifeq ($(CPU_ARCH),sh4a)
+	CPU_ARCH        = sh4
+else
+ $(CPU_ARCH) == m68k, ppc, ia64, sparc, s390, s390x, mips, sh3, sh4
+	CPU_ARCH	= $(CPU_ARCH)
+endif
+endif
+endif
+endif
+endif
+endif
+endif
+endif
+
+
+ifeq ($(OS_RELEASE),2.0)
+	OS_REL_CFLAGS	+= -DLINUX2_0
+	MKSHLIB		= $(CC) -shared -Wl,-soname -Wl,$(@:$(OBJDIR)/%.so=%.so) $(RPATH)
+	ifdef MAPFILE
+		MKSHLIB += -Wl,--version-script,$(MAPFILE)
+	endif
+	PROCESS_MAP_FILE = grep -v ';-' $< | \
+         sed -e 's,;+,,' -e 's; DATA ;;' -e 's,;;,,' -e 's,;.*,;,' > $@
+endif
+
+OS_CFLAGS	= $(DSO_CFLAGS) $(OS_REL_CFLAGS) $(ARCHFLAG) -Wno-switch -pipe -ffunction-sections -fdata-sections -DLINUX -Dlinux -DHAVE_STRERROR
+OS_LIBS	= -lpthread -ldl -lc
+
+ARCH		= linux
+
+DSO_CFLAGS		= -fPIC
+DSO_LDOPTS		= -shared $(ARCHFLAG) -Wl,--gc-sections
+# The linker on Red Hat Linux 7.2 and RHEL 2.1 (GNU ld version 2.11.90.0.8)
+# incorrectly reports undefined references in the libraries we link with, so
+# we don't use -z defs there.
+ZDEFS_FLAG		= -Wl,-z,defs
+DSO_LDOPTS		+= $(if $(findstring 2.11.90.0.8,$(shell ld -v)),,$(ZDEFS_FLAG))
+LDFLAGS		+= $(ARCHFLAG)
+
+#
+# Always set CPU_TAG on Linux.
+#
+CPU_TAG = _$(CPU_ARCH)
+
+#
+# On Linux 2.6 or later, build libfreebl3.so with no NSPR and libnssutil3.so
+# dependencies by default.  Set FREEBL_NO_DEPEND to 0 in the environment to
+# override this.
+#
+#ifneq ($(OS_TARGET),Android)
+#ifeq (2.6,$(firstword $(sort 2.6 $(OS_RELEASE))))
+#ifndef FREEBL_NO_DEPEND
+#FREEBL_NO_DEPEND = 1
+#FREEBL_LOWHASH = 1
+#endif
+#endif
+#endif
+
+USE_SYSTEM_ZLIB = 1
+ZLIB_LIBS = -lz
+
+OS_REL_CFLAGS   += -DLINUX2_1
+MKSHLIB         = $(CC) $(DSO_LDOPTS) -Wl,-soname -Wl,$(@:$(OBJDIR)/%.so=%.so) $(RPATH)
+
+ifdef MAPFILE
+	MKSHLIB += -Wl,--version-script,$(MAPFILE)
+endif
+PROCESS_MAP_FILE = grep -v ';-' $< | \
+        sed -e 's,;+,,' -e 's; DATA ;;' -e 's,;;,,' -e 's,;.*,;,' > $@
